irc/api.go

```diff
@@ -222,3 +221,134 @@ func (a *ergoAPI) handleAccountDetails(w http.ResponseWriter, r *http.Request) {
 
 	a.writeJSONResponse(response, w, r)
 }
+
+type apiNsInfoRequest struct {
+	Nick string `json:"nick"`
+}
+
+type apiNsInfoResponse struct {
+	apiGenericResponse
+	AccountName  string   `json:"accountName,omitempty"`
+	RegisteredAt string   `json:"registeredAt,omitempty"`
+	Channels     []string `json:"channels,omitempty"`
+}
+
+func (a *ergoAPI) handleNsInfo(w http.ResponseWriter, r *http.Request) {
+	var request apiNsInfoRequest
+	if err := a.decodeJSONRequest(&request, w, r); err != nil {
+		return
+	}
+
+	var response apiNsInfoResponse
+
+	if request.Nick != "" {
+		// Look up the account associated with this nick
+		accountName := a.server.accounts.NickToAccount(request.Nick)
+		if accountName == "" {
+			response.Success = false
+			a.writeJSONResponse(response, w, r)
+			return
+		}
+		// Load the account details
+		accountData, err := a.server.accounts.LoadAccount(accountName)
+		if err != nil {
+			response.Success = false
+			a.writeJSONResponse(response, w, r)
+			return
+		}
+
+		// Get the channels registered to this account
+		channels := a.server.channels.ChannelsForAccount(accountName)
+
+		// Populate the response
+		response.Success = true
+		response.AccountName = accountData.Name
+		response.RegisteredAt = accountData.RegisteredAt.Format("Mon, 02 Jan 2006 15:04:05 UTC")
+		response.Channels = channels
+	} else {
+		response.Success = false
+		response.ErrorCode = "INVALID_REQUEST"
+	}
+
+	a.writeJSONResponse(response, w, r)
+}
+
+type apiAccountListResponse struct {
+	apiGenericResponse
+	Accounts   []apiAccountDetailsResponse `json:"accounts"`
+	TotalCount int                         `json:"totalCount"`
+}
+
+func (a *ergoAPI) handleAccountList(w http.ResponseWriter, r *http.Request) {
+	var response apiAccountListResponse
+
+	// Get all account names
+	accounts := a.server.accounts.AllNicks()
+	response.TotalCount = len(accounts)
+
+	// Load account details
+	response.Accounts = make([]apiAccountDetailsResponse, len(accounts))
+	for i, account := range accounts {
+		accountData, err := a.server.accounts.LoadAccount(account)
+		if err != nil {
+			response.Accounts[i] = apiAccountDetailsResponse{
+				apiGenericResponse: apiGenericResponse{
+					Success: false,
+					Error:   err.Error(),
+				},
+			}
+			continue
+		}
+
+		response.Accounts[i] = apiAccountDetailsResponse{
+			apiGenericResponse: apiGenericResponse{
+				Success: true,
+			},
+			AccountName: accountData.Name,
+			Email:       accountData.Settings.Email,
+		}
+	}
+
+	response.Success = true
+	a.writeJSONResponse(response, w, r)
+}
+
+type StatusResponse struct {
```

Comment:
This should be named `apiStatusResponse` to avoid confusion with other types in this namespace